### DISCUSSION TITLE
Serve Next Image cache from cdn

### DISCUSSION THREAD

**Discussion by 11koukou** [vercel/next.js#36676] [Help]
Hello, 

I would like to ask if and how is possible to serve the images that Next creates in .next/cache folder from a CDN. I know that I can use a custom loader provided from an image hosting service which but I prefer using  next default loader. The reason is that next has a great image processing functionality and of course, it's free! Why pay extra for an external image service/resizing ? All I would like to do is serve the next Images from a CDN just like I can do with the static css and js files using "asset prefix" in next.config.js.

Thank you very much

**Comment by icyJoseph** [comment]
Do you want to use the next image optimizer as a service for others to use? Or do you want to use a CDN, and the CDN should in turn reach out for next.js? I am a bit confused here.

Next.js uses Sharp when it is available, otherwise, Squoosh is used, so I guess you could make an endpoint that uses Sharp to resize images (independently of next.js).

Again, I am not sure I follow what you want to do, perhaps you could draw it in https://excalidraw.com/?

**Reply by 11koukou** [reply]
When I start my nextjs app, it serves the images from the folder .next/cache, right? What I am asking is if I can serve those Images from a cdn, without using another/custom loader. Is tha possible? That happens as well with the "asset prefix" addition in the next.config.js file for serving static assetcs like.js and .css files from a cdn.

Thank you

**Comment by egoist** [comment]
I'd love to have this too. My app runs in a serverless envrionment, i.e. .next/cache is not persisted, so I would like to store and retrieve the image cache from sth like an S3 bucket.

**Comment by farzadabroudi** [comment]
I have this problem too, does anyone have an idea?

**Comment by eybel** [comment]
Any idea on this?

**Comment by fingerartur** [comment]
Answered in https://github.com/vercel/next.js/issues/33488
